File: Python/2017/Favourite Times.py
# J4

minutes = int(input())
count = 0
time = 1200

def main():
    global minutes, count, time
    for i in range(minutes):
        # add one to the time
        time += 1

        # checks to adjust timing if necessary (to next hour)
        if time == 1260:
            time = 100
        for a in range(1,12):
            if time == (a*100) + 60:
                time = (a+1) * 100
        
        time = str(time) # converts time to string to check length and do subscripts for comparisons

        # only two possibilities: time has 3 difits or 4 digits

        # if the time has 3 digits, checks if it's an arithmetic sequenc
        if len(time) == 3:
            x = int(time[0])
            y = int(time[1])
            z = int(time[2])
            if z - y == y - x:
                count += 1
            else:
                pass

        # if the time has 4 digits, checks if it's an arithmetic sequenc
        else:
            w = int(time[0])
            x = int(time[1])
            y = int(time[2])
            z = int(time[3])
            if z - y == y - x == x - w:
                count += 1
            else:
                pass

        # converting time back into an integer
        time = int(time)


if __name__ == "__main__":
    main()
    print(count)

File: C++/2017/Favourite Times.cpp
// J4

#include <iostream>
using namespace std;

int numDigits(int x){
    return to_string(x).length();
}

bool threeDigitCheck(int t) {
    string time = to_string(t);
    int x = (int)time[0] - 48;
    int y = (int)time[1] - 48;
    int z = (int)time[2] - 48;

    if (x + 1 == y && y + 1 == z) return true;

    return false;

}


bool fourDigitCheck(int t) {
    string time = to_string(t);
    int w = (int)time[0] - 48;
    int x = (int)time[1] - 48;
    int y = (int)time[2] - 48;
    int z = (int)time[3] - 48;

    if (w + 1 == x && x + 1 == y && y + 1 == z) return true;

    return false;

}


int main() {
    int minutes, count = 0, time=1200;

    cin >> minutes;

    for (int i = 0; i < minutes; i++) {
        time += 1;
        if (time == 1260) {
            time = 100;
        }
        for (int a = 1; a < 12; a++) {
            if (time == (a * 100) + 60) {
                time = (a + 1) * 100;
            }
            else {
                continue;
            }
        }
        if (numDigits(time) == 3) {
             if (threeDigitCheck(time)) count += 1;
        }
        else {
            if (fourDigitCheck(time)) count += 1;
        }
    }

    cout << count << endl;
}